### DISCUSSION TITLE
Is there  java manual sdk instead of auto-instrument way?

### DISCUSSION THREAD

**Discussion by liu1028** [apache/skywalking#9978] [Q&A] [answered]
It's the best way to use auto instrument strategy to observe application perf at most cases in java world. But in case of tracing 
 low-level storage system, such as elasticsearch kernel, it may be harmful to ES's performance. So I think providing java's manual sdk to give developer more control, such as OpenTelemetry SDK,  is a good way. Note that manual sdk means no sniffer, just use sdk api to collect directly and send to oapServer.

**Comment by wu-sheng** [comment, marked as answer]
No, there isn't.

**Reply by liu1028** [reply]
Could you please support this？May some infrastrucature java's applications(such as mq kernel, es kernel) need manual sdk indeed. It can enrich skywalking's ecosystem.

**Reply by wu-sheng** [reply]
If you feel this is useful, you could contribute this, starting from an official proposal.
There is no we can or can't. That is not how open source works.
